### PR TITLE
[Renesas]Add ra6m4-cpk lvgl for spi-lcd support

### PR DESCRIPTION
**Pull request by Rbb666**
## 拉取/合并请求描述：(PR description)

[
Add ra6m4-cpk lvgl for spi-lcd support
]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author        Notes
+ * 2023-02-23     Rbb666        First version
+ */
+#include <rtdevice.h>
+
+#ifdef BSP_USING_SPI_LCD
```
Comment (mysterywolf) [inline bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c:12]: 9341的驱动已经上传了好几遍了 可否抽出来做成一个软件包？

另外rt-thread的SPI接口新增加了新的接口，建议使用这个接口进行适配 https://github.com/RT-Thread/rt-thread/blob/master/components/drivers/spi/spi_core.c#L47

可以参见STM32的驱动，rt_spi_bus_attach_device这个API会被抛弃掉

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c (hunk 1 above)
Comment (mysterywolf) [reply bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c:12]: 可以参考一下这个diff： https://github.com/mysterywolf/rt-thread/commit/2d7ccff47967e46cb2ed9bedc3f0ec4aa8c6651f

Comment (Rbb666) [reply bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c:12]: > 

了解，后面我做成软件包吧

**Review by mysterywolf — APPROVED** [review]

File: bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c (hunk 1 above)
Comment (mysterywolf) [reply bsp/renesas/ra6m4-cpk/board/ports/ili9341/lcd_ili9341.c:12]: 嗯嗯 可以后续再把驱动做成软件包

File: bsp/renesas/libraries/HAL_Drivers/drv_spi.c [hunk 2]
```diff
@@ -101,7 +101,6 @@ static spi_bit_width_t ra_width_shift(rt_uint8_t data_width)
 static rt_err_t ra_write_message(struct rt_spi_device *device, const void *send_buf, const rt_size_t len)
 {
     RT_ASSERT(device != NULL);
-    RT_ASSERT(device->parent.user_data != NULL);
```
Comment (Guozhanxin) [inline bsp/renesas/libraries/HAL_Drivers/drv_spi.c:104]: 现在这个成员允许为空了吗？

**Review by Guozhanxin — APPROVED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_spi.c (hunk 2 above)
Comment (Rbb666) [reply bsp/renesas/libraries/HAL_Drivers/drv_spi.c:104]: enen

**Pull request merged**